src/providers/oracle/ocispatial/qsql_ocispatial.cpp

```diff
@@ -2938,7 +2882,8 @@ bool QOCISpatialCols::convertToWkb( QVariant &v, int index )
         points << Point( x1, y1 );
         points << Point( x0, y1 );
         points << Point( x0, y0 );
-        currentPartWkbType = WKBPolygon;
+        if ( currentPartWkbType == WKBUnknown )
+          currentPartWkbType = WKBPolygon;
```

Comment:
https://docs.oracle.com/database/121/SPATL/sdo_geometry-object-type.htm#SPATL492 for the ETYPE=1003/2003 and n == 3 indeed points to https://docs.oracle.com/database/121/SPATL/three-dimensional-optimized-rectangles.htm#SPATL473 . But I think that instead of a 5 point rectangle that would be a 3D box for nDims==3. Not sure for the circle. I'd be tempted to leave those edge cases for later.

Reply:
I'm agree to let the way it is.